src/writer.jl

```diff
@@ -338,10 +338,18 @@ function render(io::IO, mime::MIME"text/plain", node::Documenter.MarkdownAST.Nod
     end
 end
 
+function sanitized_anchor_label(anchor)
+    # vitepress doesn't like special markdown characters in the id slug, so we just remove them.
+    # it seems unlikely to get conflicts with other slugs this way, and escaping the characters with
+    # backslashes did not make the slugs work correctly in vitepress, either
+    label = Documenter.anchor_label(anchor)
+    return replace(label, r"[\[\]\(\)*]" => "")
+end
+
+
 function render(io::IO, mime::MIME"text/plain", node::Documenter.MarkdownAST.Node, docs::Documenter.DocsNode, page, doc; kwargs...)
-    # @infiltrate
     open_txt = get(page.globals.meta, :CollapsedDocStrings, false) ? "" : "open"
-    anchor_id = Documenter.anchor_label(docs.anchor)
+    anchor_id = sanitized_anchor_label(docs.anchor)
```

Comment:
```suggestion
        # Vitepress doesn't like Markdown in slugs, 
        # so we sanitize the output of `Documenter.anchor_label`
        anchor_id = sanitized_anchor_label(docs.anchor)
```

Reply:
that comment is in the helper function, which I think is also ok